### PR TITLE
Experimental tests [WIP]

### PR DESCRIPTION
**Pull request by foxish**
The steps to run the test would be:

0. `helm delete tf-test --purge`
1. `helm install tf-training-chart/ -n tf-test --wait`
2. `helm test tf-test`
3. `kubectl logs cmle-basic-test`

### CODE REVIEW

File: tf-training-chart/templates/operator-deployment.yaml [hunk 1]
```diff
@@ -0,0 +1,27 @@
+apiVersion: extensions/v1beta1
```
Comment (foxish) [inline tf-training-chart/templates/operator-deployment.yaml:1]: This file is copied as is from the images/tf-operator directory.

File: tf-training-chart/templates/basic-test.yaml [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+metadata:
+  name: cmle-basic-test
+  annotations:
+    "helm.sh/hook": test-success
+spec:
+  containers:
+    - name: basic-test
+      image: python:latest
+      command: ["/bin/sh","-c"]
+      args: ["wget -NP /usr/bin https://storage.googleapis.com/kubernetes-release/release/$(curl -s https://storage.googleapis.com/kubernetes-release/release/stable.txt)/bin/linux/amd64/kubectl
```
Comment (jlewi) [inline tf-training-chart/templates/basic-test.yaml:12]: Is there a closing parenthesis missing at the end here?

File: tf-training-chart/templates/basic-test-config.yaml [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+data:
+  run.py: |-
+    #! /usr/bin/python
+    from subprocess import call
+
+    def test_trivial():
+        assert "a" == "a"
+
+    def test_create():
+        return_code = call("kubectl delete -f https://raw.githubusercontent.com/jlewi/mlkube.io/master/examples/tf_job.yaml", shell=True)
```
Comment (jlewi) [inline tf-training-chart/templates/basic-test-config.yaml:14]: Do you delete it first to make sure there is no result from a previous test running?

File: tf-training-chart/templates/basic-test.yaml (hunk 2 above)
Comment (foxish) [reply tf-training-chart/templates/basic-test.yaml:12]: No, it's one long string of `args` and ends 3 lines below. I did it this way for brevity. Some other examples use init containers to download dependencies in a separate init-container but this way also works.

File: tf-training-chart/templates/basic-test-config.yaml (hunk 3 above)
Comment (foxish) [reply tf-training-chart/templates/basic-test-config.yaml:14]: Yes, just a hasty prototype. We could use the python client for k8s to write much better tests here, while talking to the API and everything.

**Comment by foxish** [comment]
Merging for now to test the CI job

**Pull request merged**